File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/class_casting/electronics_product.py
from product import Product

class ElectronicsProduct(Product):
    def __init__(self, name, price, category, brand, warranty_years):
        super().__init__(name, price, category)
        self.brand = brand
        self.warranty_years = warranty_years

    def display_info(self):
        super().display_info()
        print(f"Brand: {self.brand}, Warranty Years: {self.warranty_years}")

    def extend_warranty(self, additional_years):
        print(f"Extending warranty of {self.brand} {self.name} by {additional_years} years")
        self.warranty_years += additional_years

File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/ClassCasting/ElectronicsProduct.java
class ElectronicsProduct extends Product {
    String brand;
    int warrantyYears;

    public ElectronicsProduct(String name, double price, String category, String brand, int warrantyYears) {
        super(name, price, category);
        this.brand = brand;
        this.warrantyYears = warrantyYears;
    }

    public void displayInfo() {
        super.displayInfo();
        System.out.println("Brand: " + brand + ", Warranty Years: " + warrantyYears);
    }

    public void extendWarranty(int additionalYears) {
        System.out.println("Extending warranty of " + brand + " " + name + " by " + additionalYears + " years");
        this.warrantyYears += additionalYears;
    }
}